title: "TGATAA"
---
author: "David C. King"
date: "2022-10-18"
output: html_document
---

```{r setup, include=FALSE, }
knitr::opts_chunk$set(echo = TRUE)

library(Biostrings)
library(GenomicRanges)
library(tidyverse)
```


```{r data}
promoter_data = readRDS('~/work/ELT-2-ChIP-revision/David/01_promoters/03_output/promoter_data/all.df.rds') 
```

```{r read-data-ELT-2}
stages = list(LE='LE',L1='L1',L3='L3')
elt2.peak.data =  lapply(stages, function(stage) {
  IDR_peaks.agg = read.table(sprintf("../01_promoters/01_input/ELT2_%s_combined_IDR.df",stage)) # peaks with signal agg
  IDR_peaks.agg$V4 = NULL
  IDR_peaks.agg$V5 = NULL
  IDR_peaks.agg$V6 = NULL
  IDR_peaks.agg$V8 = NULL
  colnames(IDR_peaks.agg) <- c("chrom", "start","end","intensity",
                               "nlogq","offset","signal.mean","signal.max")
  gr.IDR = makeGRangesFromDataFrame(IDR_peaks.agg,
                                    keep.extra.columns = T)
  seqinfo(gr.IDR) <- Seqinfo(genome="ce11")
  gr.IDR$signal.sum = width(gr.IDR )*gr.IDR$signal.mean
  return(gr.IDR)
})
```

## Rob data

```{r read-rob-merged}
# david-reader.R
rob = read_rob_all_merged() %>% dplyr::select(-starts_with("pvalue."),-starts_with("lfcSE."))
glimpse(rob)
```

```{r combine-elt-2-rnaseq}
merge = right_join(rob, elt2.data, by = "WBGeneID")
# glimpse(merge)

# Left-over genes are NA in alldata. Give them a label
merge$embryo_int_exp[is.na(merge$embryo_int_exp)] = 'ftr'
merge$L1_int_exp[is.na(merge$L1_int_exp)] = 'ftr'
merge$L3_int_exp[is.na(merge$L3_int_exp)] = 'ftr'

merge$embryo.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$LE_bound, "bound", "unbound"), merge$embryo_int_exp)
merge$L1.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$L1_bound, "bound", "unbound"), merge$L1_int_exp)
merge$L3.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$L3_bound, "bound", "unbound"), merge$L3_int_exp)

# fix a name 
merge %<>% dplyr::rename(L3.log_chip_signal_mean = L3_log.chip_signal_mean)
```

```{r log2-chip-to-log10, eval=FALSE}
merge %<>% dplyr::mutate(LE.log_chip_signal_mean = LE.log_chip_signal_mean*log(2,10),
                  LE.log_chip_signal_max  = LE.log_chip_signal_max*log(2,10),
                  L1.log_chip_signal_mean = L1.log_chip_signal_mean*log(2,10),
                  L1.log_chip_signal_max  = L1.log_chip_signal_max*log(2,10),
                  L3.log_chip_signal_mean = L3.log_chip_signal_mean*log(2,10),
                  L3.log_chip_signal_max  = L3.log_chip_signal_max*log(2,10)
                  )
```

```{r choose-columns}

alldata = merge %>% dplyr::select(ends_with("log_chip_signal_mean"), 
                        starts_with("rlogc."),
                        starts_with("log2FoldChange."),
                        ends_with(".boundEnrichedStatus"),
                        ends_with("_int_exp"),
                        ends_with("_bound"),
                        din.status.description,
                        wikigene_name,
                        WBGeneID)

alldata$embryo_int_exp %<>% factor(levels=c("equal","depleted","enriched","ftr"))
alldata$L1_int_exp %<>% factor(levels=c("equal","depleted","enriched","ftr"))
alldata$L3_int_exp %<>% factor(levels=c("equal","depleted","enriched","ftr"))


```

```{r change-back-dirs}
setwd(oldwd)
```

```{r pattern-matching}

matches_to_granges = function(matchobj) {

  glist = list()
  for (chromosome_name in names(matchobj)) {
    glist[[chromosome_name]] = GRanges(seqnames=chromosome_name,                                    ranges=matchobj[[chromosome_name]],
                              seqinfo = Seqinfo(genome="ce11"))
  }
  unlist(GRangesList(glist) )
    
}

```


```{r sequence-data}

# "/Users/david/work/ELT-2-ChIP-revision/David/03_TGATAA"
proj_root = "../.."
genome.seq = readDNAStringSet("../../DATA/sequence/ce11_no_mito.fa")

david_promoter_dir = file.path(proj_root,"David/01_promoters/03_output")

promoters1200 = read.table(
  file.path(
    david_promoter_dir,
    "filtered.promoters.minus1000_plus200.df"))

promoters1200$V8 = NULL
promoters1200$V9 = NULL

colnames(promoters1200) <- c("chrom", "start", "end", "span", "strand", "wbid", "name")
gr.promoters = makeGRangesFromDataFrame(promoters1200, keep.extra.columns = T)
```

```{r TGATAA-matches}

genome.TGATAA = vmatchPattern('TGATAA', genome.seq)
genome.WGATAR = vmatchPattern('WGATAR', genome.seq, fixed=FALSE)

granges.TGATAA = matches_to_granges(genome.TGATAA)
granges.WGATAR = matches_to_granges(genome.WGATAR)

```

```{r ELT-2-peak-locations}

LE_path = file.path(proj_root, "David/01_promoters/01_input/ELT2_LE_combined_IDR.df")
L1_path = file.path(proj_root, "David/01_promoters/01_input/ELT2_L1_combined_IDR.df")
L3_path = file.path(proj_root, "David/01_promoters/01_input/ELT2_L3_combined_IDR.df")

LE = read.table(LE_path)[,1:3]
L1 = read.table(L1_path)[,1:3]
L3 = read.table(L3_path)[,1:3]
ELT2.bystage = GRangesList(list(
    LE = makeGRangesFromDataFrame(LE, seqnames.field = "V1", 
                   start.field = "V2", 
                   end.field = "V3", seqinfo = Seqinfo(genome="ce11")),
    L1 = makeGRangesFromDataFrame(L1, seqnames.field = "V1", 
                   start.field = "V2", 
                   end.field = "V3", seqinfo = Seqinfo(genome="ce11")),
    L3 = makeGRangesFromDataFrame(L3, seqnames.field = "V1", 
                   start.field = "V2", 
                   end.field = "V3", seqinfo = Seqinfo(genome="ce11"))))

```


```{r number-of-tgataa}
length(granges.TGATAA) 
# [1] 52912

lapply(ELT2.bystage,length)
```

```{r}
for (stage in names(ELT2.bystage)) {
  ELT2.bystage[[stage]]$TGATAA.count = countOverlaps(ELT2.bystage[[stage]], granges.TGATAA)
  ELT2.bystage[[stage]]$WGATAR.count = countOverlaps(ELT2.bystage[[stage]], granges.WGATAR)
}
lapply(ELT2.bystage, function(x) {mean(x$TGATAA.count)})
lapply(ELT2.bystage, function(x) {mean(x$WGATAR.count)})

```

