---
author: "David C. King"
date: "2022-10-18"
title: "TGATAA"
output: powerpoint_presentation
---

```{r setup, include=FALSE, }
knitr::opts_chunk$set(echo = FALSE)

library(Biostrings)
library(GenomicRanges)
library(tidyverse)
```

# data

```{r data}
promoter_data = readRDS('~/work/ELT-2-ChIP-revision/David/01_promoters/03_output/promoter_data/all.df.rds') 
promoter_gr = makeGRangesFromDataFrame(promoter_data, keep.extra.columns = T, seqinfo=Seqinfo(genome="ce11"))
```

# pattern-matching

```{r pattern-matching}

matches_to_granges = function(matchobj, genome.seq) {
  # get the actual sequence matches
  step1 = extractAllMatches(genome.seq[[1]], matchobj)
  step2 = as.character(step1, check.limits = FALSE)
  step3 = split(step2, names(step2))
  glist = list()
  for (chromosome_name in names(matchobj)) {
    gr = GRanges(seqnames=chromosome_name,                                                                     ranges=matchobj[[chromosome_name]], 
                 seqinfo = Seqinfo(genome="ce11"))
  
    gr$sequence = step3[[chromosome_name]]
    glist[[chromosome_name]] = gr
  }
  unlist(GRangesList(glist) )
    
}

```

# sequence-data

```{r sequence-data}

# "/Users/david/work/ELT-2-ChIP-revision/David/03_TGATAA"
proj_root = "../.."
genome.seq = readDNAStringSet("../../DATA/sequence/ce11_no_mito.fa")


gene_body_masks = makeGRangesFromDataFrame(
  read.table(
    file.path("..", 
              proj_root, 
              "ELT2-Metanalysis/DATA/HOT/gene_body_masks.bed")),
    seqnames.field = "V1", 
    start.field = "V2", 
    end.field = "V3", seqinfo = Seqinfo(genome="ce11")
  )
```

# TGATAA-matches
```{r GRvmatchPattern}
GRvmatchPattern = function(pattern, subject, max.mismatch = 0, min.mismatch = 0, 
    with.indels = FALSE, fixed = TRUE, algorithm = "auto", ...) {
    ByPos_MIndex.obj = vmatchPattern(pattern, 
                          subject, 
                          max.mismatch = max.mismatch, 
                          min.mismatch = min.mismatch, 
                          with.indels = with.indels, 
                          fixed = fixed, 
                          algorithm = algorithm, ...)
    ir = unlist(ByPos_MIndex.obj) # IRanges object 
    return(
      GRanges(
          seqnames=names(ir), 
          ranges=ir, 
          seqinfo = Seqinfo(genome="ce11"))
)
}
```

```{r GATA-matches}
# scan all WGATAR subseqs, convert to GRanges, and mask out gene bodies
granges.TGATAA = subsetByOverlaps(GRvmatchPattern('TGATAA', genome.seq), 
                                  gene_body_masks)
granges.WGATAR = subsetByOverlaps(GRvmatchPattern('WGATAR', genome.seq, fixed =
                                                    FALSE),
                                  
                                  gene_body_masks)
granges.AGATAA = subsetByOverlaps(GRvmatchPattern('AGATAA', genome.seq),
                                  gene_body_masks)
granges.GGATAA = subsetByOverlaps(GRvmatchPattern('GGATAA', genome.seq),
                                  gene_body_masks)
granges.GGATAG = subsetByOverlaps(GRvmatchPattern('GGATAG', genome.seq),
                                  gene_body_masks)
granges.AGATAG = subsetByOverlaps(GRvmatchPattern('AGATAG', genome.seq),
                                  gene_body_masks)
granges.TGATAG = subsetByOverlaps(GRvmatchPattern('TGATAG', genome.seq),
                                  gene_body_masks)
```

```{r}
table(seqnames(granges.TGATAA)) %>% addmargins(1) %>% knitr::kable(label = "TGATAAs ")
table(seqnames(granges.WGATAR)) %>% addmargins(1) %>% knitr::kable(label = "WGATARs ")



table(seqnames(granges.TGATAA)) %>% addmargins(1) %>% knitr::kable(label = "TGATAAs masked")


table(seqnames(granges.WGATAR)) %>%
  addmargins(1) %>% 
  knitr::kable(label = "WGATARs ")
```

# ELT-2-peak-locations

```{r ELT-2-peak-locations}

LE_path = file.path(proj_root, "David/01_promoters/01_input/ELT2_LE_combined_IDR.df")
L1_path = file.path(proj_root, "David/01_promoters/01_input/ELT2_L1_combined_IDR.df")
L3_path = file.path(proj_root, "David/01_promoters/01_input/ELT2_L3_combined_IDR.df")

LE = read.table(LE_path)[,1:3]
L1 = read.table(L1_path)[,1:3]
L3 = read.table(L3_path)[,1:3]
ELT2.bystage = GRangesList(list(
    LE = makeGRangesFromDataFrame(LE, seqnames.field = "V1", 
                   start.field = "V2", 
                   end.field = "V3", seqinfo = Seqinfo(genome="ce11")),
    L1 = makeGRangesFromDataFrame(L1, seqnames.field = "V1", 
                   start.field = "V2", 
                   end.field = "V3", seqinfo = Seqinfo(genome="ce11")),
    L3 = makeGRangesFromDataFrame(L3, seqnames.field = "V1", 
                   start.field = "V2", 
                   end.field = "V3", seqinfo = Seqinfo(genome="ce11"))))
addmargins(table(seqnames(ELT2.bystage)),2)


ELT2.bystage.masked = GRangesList(lapply(as.list(ELT2.bystage),
          function(stage.gr){
            subsetByOverlaps(stage.gr,gene_body_masks)
          }) )
addmargins(table(seqnames(ELT2.bystage.masked)),2)
```

# number-of-peaks-with-tgataa

```{r number-of-peaks-with-tgataa}
peaks.tgataa = GRangesList(lapply(as.list(ELT2.bystage.masked),
          function(stage.gr){
            subsetByOverlaps(stage.gr,granges.TGATAA)
          }) )
#lapply(peaks.tgataa, length)
#lapply(ELT2.bystage.masked, length)

unlist(lapply(peaks.tgataa, length))/unlist(lapply(ELT2.bystage.masked, length)) 
```

# tgataas-that-are-bound

```{r tgataas-that-are-bound}
tgataa.w.peaks = GRangesList(lapply(as.list(ELT2.bystage.masked),
          function(stage.gr){
            subsetByOverlaps(granges.TGATAA, stage.gr)
          }) )

# length(granges.TGATAA.masked)
# unlist(lapply(tgataa.w.peaks, length))
unlist(lapply(tgataa.w.peaks, length))/length(granges.TGATAA)
```

# number-of-peaks-with-wgatar
```{r get-motif-profile}
motifProfile = function(granges.query) {
  granges.query$TGATAA = countOverlaps(granges.query, granges.TGATAA)
  granges.query$AGATAA = countOverlaps(granges.query, granges.AGATAA)
  granges.query$AGATAG = countOverlaps(granges.query, granges.AGATAG)
  granges.query$TGATAG = countOverlaps(granges.query, granges.TGATAG)
  granges.query
}

peaks.wgatar.profile = GRangesList(lapply(as.list(ELT2.bystage.masked),
          function(stage.gr){
            motifProfile(stage.gr)
          }) )

a=mcols(peaks.wgatar.profile$LE) %>% as.data.frame() %>% colSums() 
b=mcols(peaks.wgatar.profile$L1) %>% as.data.frame() %>% colSums() 
c=mcols(peaks.wgatar.profile$L3) %>% as.data.frame() %>% colSums() 
sequence.profiles = rbind(a,b,c)
rownames(sequence.profiles) <- c("LE","L1","L3")
dimnames(sequence.profiles) <- list(stage=rownames(sequence.profiles),sequence=colnames(sequence.profiles))
sequence.profiles.long = sequence.profiles %>% reshape2::melt(value.name="count")
ggplot(sequence.profiles.long, aes(x=sequence, y=count)) + geom_bar(stat="identity") + facet_wrap(~stage) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```


```{r number-of-peaks-with-wgatar}
peaks.wgatar = GRangesList(lapply(as.list(ELT2.bystage.masked),
          function(stage.gr){
            subsetByOverlaps(stage.gr,granges.WGATAR)
          }) )
# print("number of peaks with WGATAR")
# lapply(peaks.wgatar, length)
# print("total numbers of peaks")
# lapply(ELT2.bystage.masked, length)
# 
# print("fraction of ELT-2 peaks with WGATAR")
unlist(lapply(peaks.wgatar, length))/unlist(lapply(ELT2.bystage.masked, length))
```
# WGATAR-that-are-bound

```{r wgatars-that-are-bound}
wgatar.w.peaks = GRangesList(lapply(as.list(ELT2.bystage.masked),
          function(stage.gr){
            subsetByOverlaps(granges.WGATAR, stage.gr)
          }) )

# length(granges.WGATAR.masked)
# unlist(lapply(wgatar.w.peaks, length))
unlist(lapply(wgatar.w.peaks, length))/length(granges.WGATAR)
```

# promoters

```{r promoters}
promoter_gr$TGATAA = countOverlaps(promoter_gr, granges.TGATAA)
promoter_gr$TGATAA.bound.LE = countOverlaps(promoter_gr, tgataa.w.peaks$LE)
promoter_gr$TGATAA.bound.L1 = countOverlaps(promoter_gr, tgataa.w.peaks$L1)
promoter_gr$TGATAA.bound.L3 = countOverlaps(promoter_gr, tgataa.w.peaks$L3)
promoter_df = as.data.frame(promoter_gr)
```

# LE

```{r LE}

promoter_df %>% mutate(embryo_enriched = !is.na(embryo_int_exp) & (embryo_int_exp == c("enriched"))) %>% group_by(LE_bound, embryo_enriched) %>% 
  summarize(n=n(),mean_TGATAA_count = mean(TGATAA), 
            mean_bound_TGATAA_count = mean(TGATAA.bound.LE),
            no_TGATAA = sum(TGATAA == 0),
            no_bound_TGATAA = sum(TGATAA.bound.LE == 0),
            have_TGATAA_but_not_bound = no_bound_TGATAA-no_TGATAA) %>% knitr::kable(label="embryo")

tmp=promoter_df %>% mutate(embryo_enriched = !is.na(embryo_int_exp) & (embryo_int_exp == c("enriched"))) 
```

# L1

```{r L1}
promoter_df %>% filter(L1_int_exp %in% c("enriched")) %>% group_by(L1_bound, L1_int_exp) %>% 
  summarize(n=n(),mean_TGATAA_count = mean(TGATAA), 
            mean_bound_TGATAA_count = mean(TGATAA.bound.L1),
            no_TGATAA = sum(TGATAA == 0),
            no_bound_TGATAA = sum(TGATAA.bound.L1 == 0),
            have_TGATAA_but_not_bound = no_bound_TGATAA-no_TGATAA) %>% knitr::kable(label="L1")

```

# L3

```{r L3}
promoter_df %>% filter(L3_int_exp %in% c("enriched")) %>% group_by(L3_bound, L3_int_exp) %>% 
  summarize(n=n(),mean_TGATAA_count = mean(TGATAA), 
            mean_bound_TGATAA_count = mean(TGATAA.bound.L3),
            no_TGATAA = sum(TGATAA == 0),
            no_bound_TGATAA = sum(TGATAA.bound.L3 == 0),
            have_TGATAA_but_not_bound = no_bound_TGATAA-no_TGATAA)%>% knitr::kable(label="L3")

```

# Dineen-classes

```{r dineen-analysis, echo=FALSE}
promoter_df %>% filter(L1_int_exp %in% c("enriched") & !is.na(din.status.description)) %>% group_by(L1_bound, din.status.description) %>% 
  summarize(n=n(),mean_TGATAA_count = mean(TGATAA), 
            mean_bound_TGATAA_count = mean(TGATAA.bound.L1),
            no_TGATAA = sum(TGATAA == 0),
            no_bound_TGATAA = sum(TGATAA.bound.L1 == 0),
            have_TGATAA_but_not_bound = no_bound_TGATAA-no_TGATAA) %>% knitr::kable()


```

# TGATAA positions mapped to promoter coordinates

```{r}
laps = findOverlapPairs(promoter_gr, tgataa.w.peaks$LE)
TSS = start(laps@first) + ifelse(strand(laps@first) == "+", 1000,200)
relative_tgataa_pos = (start(laps@second) - TSS) * ifelse(strand(laps@first) == "+", 1,-1)
```