---
title: "signal correlations"
author: "DC King - Onish lab"
date: "4/14/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message=FALSE)
library(tidyr)
library(dplyr)
library(GenomicRanges)
library(magrittr)
library(knitr) # for kable
library(ggplot2)
library(cowplot)
library(multcompView)
library(ggridges)
library(ggpubr)
source('david-reader.R')
okabe <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
options(ggplot2.discrete.color=okabe) # this doesn't work

```

## ELT-2 data

```{r read-data-ELT-2}
elt2.data = read_ELT2_binding_data(as_genomic_ranges = FALSE)
#glimpse(elt2.data)
```

## Rob data

```{r read-rob-merged}
rob = read_rob_all_merged() %>% dplyr::select(-starts_with("pvalue."),-starts_with("lfcSE."))
glimpse(rob)
```

```{r combine-elt-2-rnaseq}
merge = right_join(rob, elt2.data, by = "WBGeneID")
# glimpse(merge)

# Left-over genes are NA in alldata. Give them a label
merge$embryo_int_exp[is.na(merge$embryo_int_exp)] = 'ftr'
merge$L1_int_exp[is.na(merge$L1_int_exp)] = 'ftr'
merge$L3_int_exp[is.na(merge$L3_int_exp)] = 'ftr'

merge$embryo.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$LE_bound, "bound", "unbound"), merge$embryo_int_exp)
merge$L1.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$L1_bound, "bound", "unbound"), merge$L1_int_exp)
merge$L3.boundEnrichedStatus = sprintf("%s.%s", ifelse(merge$L3_bound, "bound", "unbound"), merge$L3_int_exp)

# fix a name 
merge %<>% dplyr::rename(L3.log_chip_signal_mean = L3_log.chip_signal_mean)
```

```{r log2-chip-to-log10, eval=FALSE}
merge %<>% dplyr::mutate(LE.log_chip_signal_mean = LE.log_chip_signal_mean*log(2,10),
                  LE.log_chip_signal_max  = LE.log_chip_signal_max*log(2,10),
                  L1.log_chip_signal_mean = L1.log_chip_signal_mean*log(2,10),
                  L1.log_chip_signal_max  = L1.log_chip_signal_max*log(2,10),
                  L3.log_chip_signal_mean = L3.log_chip_signal_mean*log(2,10),
                  L3.log_chip_signal_max  = L3.log_chip_signal_max*log(2,10)
                  )
```

```{r choose-columns}

alldata = merge %>% dplyr::select(ends_with("log_chip_signal_mean"), 
                        starts_with("rlogc."),
                        starts_with("log2FoldChange."),
                        ends_with(".boundEnrichedStatus"),
                        ends_with("_int_exp"),
                        ends_with("_bound"),
                        din.status.description,
                        WBGeneID)

```

```{r embryo-r-squared}
data = alldata
#data %<>% filter(WBGeneID != 'WBGene00016153') # negative signal, lowest point
#data %<>% filter(WBGeneID != 'WBGene00012762') # kind of hot region, highest point in "depleted"

# trim extreme .1%
q.LE.chipsig = quantile(data$LE.log_chip_signal_mean, c(1-.9995,.9995))
q.rlogc.embryo = quantile(data$rlogc.embryo, c(1-.9995,.9995),na.rm=T)
data %<>% filter(between(LE.log_chip_signal_mean, q.LE.chipsig[1],q.LE.chipsig[2]))
data %<>% filter(between(rlogc.embryo, q.rlogc.embryo[1],q.rlogc.embryo[2]))

#cor(data$LE.log_chip_signal_mean,data$rlogc.embryo,use="complete")

# methods=c("pearson","spearman")
# for (method in methods){
# d= data %>% filter(LE_bound == TRUE); cor(d$LE.log_chip_signal_mean,d$rlogc.embryo,use="complete", method=method) %>% print()
# d= data %>% filter(LE_bound == FALSE); cor(d$LE.log_chip_signal_mean,d$rlogc.embryo,use="complete", method=method) %>% print()
# d= data %>% filter(embryo_int_exp == 'ftr'); cor(d$LE.log_chip_signal_mean,d$rlogc.embryo,use="complete", method=method) %>% print()
# d = data %>% filter(embryo_int_exp == 'enriched'); cor(d$LE.log_chip_signal_mean,d$rlogc.embryo,use="complete", method=method) %>% print()
# d= data %>% filter(embryo_int_exp == 'equal'); cor(d$LE.log_chip_signal_mean,d$rlogc.embryo,use="complete", method=method) %>% print()
# d= data %>% filter(embryo_int_exp == 'depleted'); cor(d$LE.log_chip_signal_mean,d$rlogc.embryo,use="complete", method=method) %>% print()
# }


# base model (0): chip signal
mod.embryo0 = lm(rlogc.embryo ~ LE.log_chip_signal_mean, data = data)
s.mod.embryo0 = summary(mod.embryo0)

# model 1: binding
mod.embryo1 = lm(rlogc.embryo ~ LE.log_chip_signal_mean*LE_bound, data = data)
s.mod.embryo1 = summary(mod.embryo1)

embryo.explained.binding = s.mod.embryo1$r.square - s.mod.embryo0$r.squared

# model 2: enrichment
mod.embryo2 = lm(rlogc.embryo ~ LE.log_chip_signal_mean*embryo_int_exp, data = data)
s.mod.embryo2 = summary(mod.embryo2)

embryo.explained.enrichment = s.mod.embryo2$r.square - s.mod.embryo0$r.squared

# full model
mod.embryo3 = lm(rlogc.embryo ~ LE.log_chip_signal_mean*LE_bound*embryo_int_exp, data = data)
s.mod.embryo3 = summary(mod.embryo3)
embryo.explained.combined = s.mod.embryo3$r.square - s.mod.embryo0$r.squared

# fit full model 
fitted.embryo3 = predict(mod.embryo3, data)
data %<>% mutate(elt2.binding = factor(ifelse(LE_bound, "ELT-2 bound", "ELT-2 unbound"),
                                       levels = c("ELT-2 unbound","ELT-2 bound")))

anova1 = anova(mod.embryo1) 
anova2 = anova(mod.embryo2) 
anova3 = anova(mod.embryo3) 

tbl1 = cbind((anova1$`Sum Sq`)/sum(anova1$`Sum Sq`), cumsum(anova1$`Sum Sq`)/sum(anova1$`Sum Sq`))
rownames(tbl1) = rownames(anova1)
tbl2 = cbind((anova2$`Sum Sq`)/sum(anova2$`Sum Sq`), cumsum(anova2$`Sum Sq`)/sum(anova2$`Sum Sq`))
rownames(tbl2) = rownames(anova2)
tbl3 = cbind((anova3$`Sum Sq`)/sum(anova3$`Sum Sq`), cumsum(anova3$`Sum Sq`)/sum(anova3$`Sum Sq`))
rownames(tbl3) = rownames(anova3)

tbl1 %>% kable(caption="Variance in embryo/LE RLog counts explained by variance in chip signal, accounting for  ELT-2 binding", col.names = c("Variance explained", "Cumulative Variance explained"), digits=c(4))
tbl2 %>% kable(caption="Variance in embryo/LE RLog counts explained by variance in chip signal, accounting for  intestine enrichment", col.names = c("Variance explained", "Cumulative Variance explained"), digits=c(4))
tbl3 %>% kable(caption="Variance in embryo/LE RLog counts explained by variance in chip signal, accounting for  combined ELT-2 binding and intestine enrichment", col.names = c("Variance explained", "Cumulative Variance explained"), digits=c(4))

p1 = ggplot(data, 
       aes(x= LE.log_chip_signal_mean, y=rlogc.embryo)) + 
  geom_point(alpha=.5) +
  stat_smooth(#color="black", 
              method="lm", 
              formula = y~x) + 
  ggtitle(bquote("ChIP signal only," ~ R^2 == .(round(s.mod.embryo0$r.squared,3)))) +
  ylab("DESeq2 rlog counts")+ scale_color_manual(values=okabe) + guides(color="none")

# facet on binding only
p2 = ggplot(data,
       aes(x= LE.log_chip_signal_mean, y=rlogc.embryo,  color=elt2.binding,group=elt2.binding)) + 
  geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x) + 
  ylim(c(0,19)) + 
  facet_wrap(~elt2.binding ) +
  ylab("DESeq2 rlog counts") + 
  ggtitle(bquote("ChIP_signal and ELT-2 binding" ~ R^2 == .(round(s.mod.embryo1$r.squared,3))))+ scale_color_manual(values=okabe)+ guides(color="none")

# facet on enrichment only
p3 = ggplot(data,
       aes(x= LE.log_chip_signal_mean, y=rlogc.embryo, color=elt2.binding,)) + 
  geom_point(alpha=.5) +
  stat_smooth(color="black", method="lm", formula = y~x) + 
  facet_wrap(~embryo_int_exp)+ 
  ylim(c(0,19)) + 
  ggtitle(bquote("ChIP_signal and intestine enrichment status" ~ R^2 == .(round(s.mod.embryo2$r.squared,3)))) + ylab("DESeq2 rlog counts") + scale_color_manual(values=okabe)+ guides(color="none")

# both binding and intestine enrichment faceted
p4 = ggplot(data,
       aes(x= LE.log_chip_signal_mean, 
           y=rlogc.embryo, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) +
  stat_smooth(color="black", 
              method="lm", 
              formula = y~x) + 
  ylim(c(0,19)) + 
  facet_wrap(~elt2.binding + embryo_int_exp) +
  ylab("DESeq2 rlog counts") + 
  ggtitle(bquote("ChIP signal, ELT-2 binding and enrichment," 
                 ~ R^2 == .(round(s.mod.embryo3$r.squared,3))))+ scale_color_manual(values=okabe)+ guides(color="none")


p5 = ggplot(data.frame(data,fitted.embryo3), 
            aes(y=fitted.embryo3, 
                x=rlogc.embryo)) + 
  geom_point(alpha=.5) + 
  stat_smooth(#color="black", 
              method = "lm", 
              formula = y~x) + 
  ggtitle(bquote("Full model," 
                 ~ R^2 == .(round(s.mod.embryo3$r.squared,3)))) + 
  xlab("DESeq2 rlog counts") + 
  #ylim(c(0,15.5)) + 
  #facet_wrap(~elt2.binding) + 
  scale_color_manual(values=okabe)

plot_grid(p1,p2,p3,p4,p5, ncol=2, labels="AUTO"); ggsave("test.pdf", height=15,width=12)
# who's the dead boy in ftr?
ftr.lm = lm(rlogc.embryo ~ LE.log_chip_signal_mean, data %>% filter(embryo_int_exp == "ftr"))
#plot(ftr.lm) show point 10993 (10597 is high)
```


```{r smash}
# smash together the variables returned by lm, summary, and anova
lm.combo = function(formula, data, subset) {
  d = data[subset,]
  mod = lm(formula=formula, data=d)
  summ = summary(mod)
  for (name in names(s.mod.embryo0)) { 
    mod[[name]] = summ[[name]]
  }
  mod$anova = anova(mod)
  mod
}
```



```{r L1}

data = alldata
# data %<>% filter(L1_int_exp != "ftr")
#data %<>% filter(L1_bound == TRUE)
#cor(data$L1.log_chip_signal_mean,data$rlogc.L1,use="complete")

# trim extreme .1%
q.L1.chipsig = quantile(data$L1.log_chip_signal_mean, c(1-.9995,.9995))
q.rlogc.L1 = quantile(data$rlogc.L1, c(1-.9995,.9995),na.rm=T)
data %<>% filter(between(L1.log_chip_signal_mean, q.L1.chipsig[1],q.L1.chipsig[2]))
data %<>% filter(between(rlogc.L1, q.rlogc.L1[1],q.rlogc.L1[2]))


mod.L10 = lm(rlogc.L1 ~ L1.log_chip_signal_mean, data = data)
s.mod.L10 = summary(mod.L10)

mod.L11 = lm(rlogc.L1 ~ L1.log_chip_signal_mean+L1_bound, data = data)
s.mod.L11 = summary(mod.L11)

mod.L12 = lm(rlogc.L1 ~ L1.log_chip_signal_mean*L1_int_exp, data = data)
s.mod.L12 = summary(mod.L12)

mod.L13 = lm(rlogc.L1 ~ L1.log_chip_signal_mean*L1_int_exp*L1_bound, data = data)
s.mod.L13 = summary(mod.L13)

mod.L14 = lm(rlogc.L1 ~ L1.log_chip_signal_mean*L1_int_exp*L1_bound*din.status.description, data = data)
s.mod.L14 = summary(mod.L14)

anova1 = anova(mod.L11)
anova2 = anova(mod.L12)
anova3 = anova(mod.L13)
anova4 = anova(mod.L14)

fitted.L13 = predict(mod.L13, data)
data %<>% mutate(elt2.binding = factor(ifelse(LE_bound, "ELT-2 bound", "ELT-2 unbound"),
                                       levels = c("ELT-2 unbound","ELT-2 bound")))

p1=ggplot(data, 
       aes(x=L1.log_chip_signal_mean, 
           y=rlogc.L1, group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) +
  geom_smooth(color="black", method="lm", formula = y~x) + 
  ggtitle(bquote("Expression level explained by ChIP signal," ~ R^2 == .(round(s.mod.L10$r.squared,3)))) +
  ylab("DESeq2 rlog counts") + scale_color_manual(values=okabe)

p2=ggplot(data,
       aes(x= L1.log_chip_signal_mean, 
           y=rlogc.L1, 
           group=elt2.binding, 
           color=elt2.binding)) + geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x) + 
  ylim(c(0,19)) + 
  facet_wrap(~elt2.binding ) +
  ylab("DESeq2 rlog counts") + 
  ggtitle(bquote("Expression explained by ChIP_signal\nELT-2  BOUND and intestine enrichment status" ~ R^2 == .(round(s.mod.L11$r.squared,3)))) + scale_color_manual(values=okabe)

p3=ggplot(data,
       aes(x=L1.log_chip_signal_mean, 
           y=rlogc.L1, 
          # group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) + 
  stat_smooth(method="lm", 
              formula = y~x, 
              color="black") + 
  facet_wrap(~L1_int_exp) + 
  ylim(c(0,19)) + 
  ggtitle(bquote("Expression explained by ChIP_signal and \nintestine enrichment status" ~ R^2 == .(round(s.mod.L12$r.squared,3)))) + 
  ylab("DESeq2 rlog counts") + scale_color_manual(values=okabe)

p4=ggplot(data,
       aes(x= L1.log_chip_signal_mean, 
           y=rlogc.L1, 
           group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point() +
  stat_smooth(method="lm", formula = y~x, color="black") + 
  ylim(c(0,19)) + 
  facet_wrap(~elt2.binding + L1_int_exp) +
  ylab("DESeq2 rlog counts") + 
  ggtitle(bquote("Expression explained by ChIP_signal\nELT-2  BOUND and intestine enrichment status" ~ R^2 == .(round(s.mod.L13$r.squared,3)))) + 
  scale_color_manual(values=okabe)

# Full model
p5=ggplot(data.frame(data,fitted.L13), 
       aes(x=fitted.L13,
           y=rlogc.L1, 
           #group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) + 
  stat_smooth(method = "lm", 
              formula = y~x, 
              color="black") + ggtitle(bquote("Full model," ~ R^2 == .(round(s.mod.L13$r.squared,3)))) + ylab("DESeq2 rlog counts")  + scale_color_manual(values=okabe) #+ facet_wrap(~elt2.binding )



# https://stats.stackexchange.com/questions/79399/calculate-variance-explained-by-each-predictor-in-multiple-regression-using-r

# cbind(rownames(anova1), cumsum(anova1$`Sum Sq`)/sum(anova1$`Sum Sq`)) %>% kable()
# cbind(rownames(anova2), cumsum(anova2$`Sum Sq`)/sum(anova2$`Sum Sq`))%>% kable()
tbl = cbind((anova3$`Sum Sq`)/sum(anova3$`Sum Sq`), cumsum(anova3$`Sum Sq`)/sum(anova3$`Sum Sq`))
rownames(tbl) = rownames(anova3)
tbl %>% kable(caption="Variance in L1 RLog counts explained by variance in chip signal, accounting for different factors", col.names = c("Variance explained", "Cumulative Variance explained"), digits=c(5))

plot_grid(p1,p2,p3,p4,p5, ncol=2, labels="AUTO"); ggsave("L1.cor.pdf", height=15,width=12)

```

```{r L3}
data = alldata
# trim extreme .1%
q.L3.chipsig = quantile(data$L3.log_chip_signal_mean, c(1-.9995,.9995))
q.rlogc.L3 = quantile(data$rlogc.L3, c(1-.9995,.9995),na.rm=T)
data %<>% filter(between(L3.log_chip_signal_mean, q.L3.chipsig[1],q.L3.chipsig[2]))
data %<>% filter(between(rlogc.L3, q.rlogc.L3[1],q.rlogc.L3[2]))

mod.L30 = lm(rlogc.L3 ~ L3.log_chip_signal_mean, data = data)
s.mod.L30 = summary(mod.L30)

mod.L31 = lm(rlogc.L3 ~ L3.log_chip_signal_mean*L3_bound, data = data)
s.mod.L31 = summary(mod.L31)

mod.L32 = lm(rlogc.L3 ~ L3.log_chip_signal_mean*L3_int_exp, data = data)
s.mod.L32 = summary(mod.L32)

mod.L33 = lm(rlogc.L3 ~ L3.log_chip_signal_mean*L3_int_exp*L3_bound, data = data)
s.mod.L33 = summary(mod.L33)

anova1 = anova(mod.L31)
anova2 = anova(mod.L32)
anova3 = anova(mod.L33)

fitted.L33 = predict(mod.L33, data)

data %<>% mutate(elt2.binding = factor(ifelse(LE_bound, "ELT-2 bound", "ELT-2 unbound"),
                                       levels = c("ELT-2 unbound","ELT-2 bound")))

p1 = ggplot(data, 
       aes(x=L3.log_chip_signal_mean, 
           y=rlogc.L3, 
           color=elt2.binding)) + 
  geom_point(alpha=.5) +
  stat_smooth(color="black", method="lm", formula = y~x) + 
  ggtitle(bquote("Expression level explained by ChIP signal," ~ R^2 == .(round(s.mod.L30$r.squared,3)))) +
  ylab("DESeq2 rlog counts") + scale_color_manual(values=okabe)

p2 = ggplot(data,
       aes(x= L3.log_chip_signal_mean, 
           y=rlogc.L3, 
           group=elt2.binding, 
           color=elt2.binding)) + 
  geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x) + 
  ylim(c(0,19)) + facet_wrap(~elt2.binding ) +
  ylab("DESeq2 rlog counts") + 
  ggtitle(bquote("Expression explained by ChIP_signal\nELT-2  BOUND and intestine enrichment status" ~ R^2 == .(round(s.mod.L31$r.squared,3)))) + scale_color_manual(values=okabe)

p3 = ggplot(data,
       aes(x=L3.log_chip_signal_mean, 
           y=rlogc.L3, color=elt2.binding)) + 
  geom_point(alpha=.5) + 
  stat_smooth(color="black", method="lm", formula = y~x) + 
  facet_wrap(~L3_int_exp)+ ylim(c(0,19)) + 
  ggtitle(bquote("Expression explained by ChIP_signal and \nintestine enrichment status" ~ R^2 == .(round(s.mod.L32$r.squared,3)))) + ylab("DESeq2 rlog counts")  + scale_color_manual(values=okabe)

p4 = ggplot(data,
       aes(x= L3.log_chip_signal_mean, y=rlogc.L3, color=elt2.binding)) + geom_point() +
  stat_smooth(color="black", method="lm", formula = y~x) + ylim(c(0,19)) + facet_wrap(~elt2.binding + L3_int_exp) +
  ylab("DESeq2 rlog counts") + ggtitle(bquote("Expression explained by ChIP_signal\nELT-2  BOUND and intestine enrichment status" ~ R^2 == .(round(s.mod.L33$r.squared,3)))) + scale_color_manual(values=okabe)


p5 = ggplot(data.frame(data,fitted.L33), 
       aes(x=fitted.L33, y=rlogc.L3, color=elt2.binding)) + 
  geom_point(alpha=.5) + 
  stat_smooth(color="black", method = "lm", formula = y~x) + 
  ggtitle(bquote("Full model," ~ R^2 == .(round(s.mod.L33$r.squared,3)))) + 
  ylab("DESeq2 rlog counts")  + scale_color_manual(values=okabe)




# https://stats.stackexchange.com/questions/79399/calculate-variance-explained-by-each-predictor-in-multiple-regression-using-r

# cbind(rownames(anova1), cumsum(anova1$`Sum Sq`)/sum(anova1$`Sum Sq`)) %>% kable()
# cbind(rownames(anova2), cumsum(anova2$`Sum Sq`)/sum(anova2$`Sum Sq`))%>% kable()
tbl = cbind(anova3$`Sum Sq`/sum(anova3$`Sum Sq`), cumsum(anova3$`Sum Sq`)/sum(anova3$`Sum Sq`))
rownames(tbl) = rownames(anova3)

tbl %>% kable(caption="Variance in L3 RLog counts explained by variance in chip signal, accounting for different factors", col.names = c("Variance explained", "Cumulative Variance explained"), digits=c(4))


L3.individual.lms = apply(
  expand.grid(levels(data$elt2.binding), 
              unique(data$L3_int_exp)), 1, 
  function(x) {

    subset = data$elt2.binding == x[[1]] & data$L3_int_exp == x[[2]];

    lmc = NA
    if (sum(subset) > 0)
    {   
      lmc = lm.combo(rlogc.L3 ~ L3.log_chip_signal_mean,data=data[subset,])
    }

    data.frame(term1=x[[1]],
      term2=x[[2]],
      r.squared = ifelse(!any(is.na(lmc)), lmc$r.squared, NA),
      
      term1.SS = ifelse(!any(is.na(lmc)), lmc$anova$`Sum Sq`[1], NA),
      term2.SS = ifelse(!any(is.na(lmc)), lmc$anova$`Sum Sq`[2], NA),
      n=sum(subset), 
      n.finite=sum(
        is.finite(unlist(data[subset, 
                    c('rlogc.L3', 'L3.log_chip_signal_mean')]
               )))
      )
    }) %>% bind_rows()
# this doesn't work to get the partial r-squared, because you're only applying
# each submodel to its subset of the data
l = lm.combo(rlogc.L3 ~ L3.log_chip_signal_mean, data = data) 

plot_grid(p1,p2,p3,p4,p5, ncol=2, labels="AUTO"); ggsave("L3.cor.pdf", height=15,width=12)
```

```{r get-hsd-row}
getHSDRow = function(x, facCombo1, facCombo2) {
   try1 = paste(facCombo1,facCombo2, sep="-")
   message(try1)
   try2 = paste(facCombo2,facCombo1, sep="-")
   message(try2)
   row1 = as.data.frame(x)[try1,]
   row2 = as.data.frame(x)[try2,]
   row1[is.na(row1)] <- 0
   row2[is.na(row2)] <- 0
   final = row1 + row2
   rownames(final) <- try1
   final
}
```


```{r ANOVAE}
data = alldata
din.status.description = data$din.status.description

LE_bound = ifelse(data$LE_bound, "ELT2 BOUND", "ELT2 UNbound")


rlogc.embryo = data$rlogc.embryo
embryo_int_exp = data$embryo_int_exp
embryo_int_exp[ embryo_int_exp == "ftr" ] = "not passing"
embryo.aov = aov(rlogc.embryo ~ LE_bound * embryo_int_exp)
embryo.hsd = TukeyHSD(embryo.aov, conf.level = .999)
opar = par(mar=c(3,10,2,1)); plot(embryo.hsd, las=2,cex.axis=.5); par(opar)
df = data.frame(expression.level = rlogc.embryo, 
                LE_bound = LE_bound, 
                intestine_enriched=embryo_int_exp,
                levels.combined = paste(LE_bound, embryo_int_exp, sep=", "))
table(df$levels.combined)
df$levels.combined = factor(df$levels.combined, levels=rev(c(
      "ELT2 BOUND, equal",
      "ELT2 UNbound, equal",
      "ELT2 BOUND, enriched",
      "ELT2 UNbound, enriched",
      "ELT2 BOUND, not passing",
      "ELT2 UNbound, not passing",
      "ELT2 BOUND, depleted",
      "ELT2 UNbound, depleted"
)))


# get the letters
compare = function(x,y) {x<y}
comp = do.call(compare, list(embryo.hsd$`LE_bound:embryo_int_exp`[,4],.01))
comp[is.na(comp)] <- FALSE

# filter out depleted
# bad = (1:length(comp))[grepl('depleted', names(comp))]
# df %<>% filter(levels.combined != "ELT2 UNbound, depleted") %>% filter(levels.combined != "ELT2 BOUND, depleted")


#letters = multcompLetters(comp[-bad])$Letters
letters = multcompLetters(comp)$Letters
names(letters) <- sub(":", ", ", names(letters))
df$multCompLetters = letters[as.character(df$levels.combined)]

# %>% filter(combo != "FALSE depleted" 
G.LE = ggplot(df, aes(x=expression.level, y=levels.combined, fill=multCompLetters)) + geom_density_ridges2(scale=2, quantile_lines=TRUE,
                       quantile_fun=function(x,...) mean(x)) + ggtitle("ANOVA followed by Tukey Honest Significant Differences (multicomparison correction)") + ylab("Expression/ELT-2 binding") + xlab("Embryo intestine expression level") 


df.embryo = df

tbl = table(df.embryo$levels.combined) 
tbl = tbl[tbl!=0]
Ns = data.frame(group=names(tbl), val=c(tbl), x=18.5, y=1:length(tbl) + .25)

G.LE = G.LE +
  geom_text(inherit.aes = F, data=Ns, aes(y=y, x=x, label=paste("n",Ns$val,sep="=")))
G.LE
ggsave("embryo.rlogc.ridgeplot.pdf")

# L1
L1.aov = aov(rlogc.L1 ~ L1_bound * L1_int_exp, data=data)
L1_bound = ifelse(data$L1_bound, "ELT2 BOUND", "ELT2 UNbound")
rlogc.L1 = data$rlogc.L1
L1_int_exp = data$L1_int_exp
L1_int_exp[ L1_int_exp == "ftr" ] = "not passing"
L1.aov = aov(rlogc.L1 ~ L1_bound * L1_int_exp)
L1.hsd = TukeyHSD(L1.aov, conf.level = .999)
opar = par(mar=c(3,10,2,1)); plot(L1.hsd, las=2,cex.axis=.5); par(opar)
df = data.frame(expression.level = rlogc.L1, 
                L1_bound = L1_bound, 
                intestine_enriched=L1_int_exp,
                levels.combined = paste(L1_bound, L1_int_exp, sep=", "))
table(df$levels.combined)

df.depleted = df %>% filter(intestine_enriched == "depleted")
G1.L1.depleted = ggplot(df.depleted, aes(x=expression.level, 
                      y=levels.combined)) + 
  geom_density_ridges2(scale=2, quantile_lines=TRUE,
                       quantile_fun=function(x,...) mean(x))

df$levels.combined = factor(df$levels.combined, levels=rev(c(
      "ELT2 BOUND, equal",
      "ELT2 UNbound, equal",
      "ELT2 BOUND, enriched",
      "ELT2 UNbound, enriched",
      "ELT2 BOUND, not passing",
      "ELT2 UNbound, not passing",
      "ELT2 BOUND, depleted",
      "ELT2 UNbound, depleted"
)))


# get the letters
comp = do.call(compare, list(L1.hsd$`L1_bound:L1_int_exp`[,4],.01))
comp[is.na(comp)] <- FALSE

# filter out depleted
bad = (1:length(comp))[grepl('depleted', names(comp))]
#df %<>% filter(levels.combined != "ELT2 UNbound, depleted") %>% filter(levels.combined != "ELT2 BOUND, depleted")

letters.w.depleted = multcompLetters(comp)$Letters
#letters = multcompLetters(comp[-bad])$Letters
letters = letters.w.depleted
names(letters) <- sub(":", ", ", names(letters))
df$multCompLetters = letters[as.character(df$levels.combined)]

# %>% filter(combo != "FALSE depleted" 
G.L1 = ggplot(df, aes(x=expression.level, y=levels.combined, fill=multCompLetters)) + geom_density_ridges2(scale=2, quantile_lines=TRUE,
                       quantile_fun=function(x,...) mean(x)) + ggtitle("ANOVA followed by Tukey Honest Significant Differences (multicomparison correction)", subtitle="Different colors: signif. difference in means at .01")  + ylab("Expression/ELT-2 binding") + xlab("L1 intestine expression level") 


df.L1 = df

tbl = table(df.L1$levels.combined) 
tbl = tbl[tbl!=0]
Ns = data.frame(group=names(tbl), val=c(tbl), x=18.5, y=1:length(tbl) + .25)

G.L1 = G.L1 +
  geom_text(inherit.aes = F, data=Ns, aes(y=y, x=x, label=paste("n",Ns$val,sep="=")))
G.L1
ggsave("L1.rlogc.ridgeplot.pdf")

# L3

L3_bound = ifelse(data$L3_bound, "ELT2 BOUND", "ELT2 UNbound")
L3_int_exp = data$L3_int_exp
L3_int_exp[ L3_int_exp == "ftr" ] = "not passing"
rlogc.L3 = data$rlogc.L3
L3.aov = aov(rlogc.L3 ~ L3_bound * L3_int_exp)
L3.hsd = TukeyHSD(L3.aov, conf.level = .99)


opar = par(mar=c(3,15,2,1)); plot(L3.hsd, las=2,cex.axis=.5); par(opar)
df = data.frame(expression.level = rlogc.L3, 
                L3_bound = L3_bound, 
                intestine_enriched=L3_int_exp,
                levels.combined = paste(L3_bound, L3_int_exp, sep=", "))
table(df$levels.combined)


df.depleted = df %>% filter(intestine_enriched == "depleted")
G.L3.depleted = ggplot(df.depleted, aes(x=expression.level, 
                      y=levels.combined)) + 
  geom_density_ridges2(scale=2, quantile_lines=TRUE,
                       quantile_fun=function(x,...) mean(x))


df$levels.combined = factor(df$levels.combined, levels=rev(c(
      "ELT2 BOUND, equal",
      "ELT2 UNbound, equal",
      "ELT2 BOUND, enriched",
      "ELT2 UNbound, enriched",
      "ELT2 BOUND, not passing",
      "ELT2 UNbound, not passing",
      "ELT2 BOUND, depleted",
      "ELT2 UNbound, depleted"
)))


# get the letters
comp = do.call(compare, list(L3.hsd$`L3_bound:L3_int_exp`[,4],.01))
comp[is.na(comp)] <- FALSE

# filter out depleted
bad = (1:length(comp))[grepl('depleted', names(comp))]
#df %<>% filter(levels.combined != "ELT2 UNbound, depleted") %>% filter(levels.combined != "ELT2 BOUND, depleted")

letters.w.depleted = multcompLetters(comp)$Letters
#letters = multcompLetters(comp[-bad])$Letters
letters = letters.w.depleted

names(letters) <- sub(":", ", ", names(letters))
df$multCompLetters = letters[as.character(df$levels.combined)]

# %>% filter(combo != "FALSE depleted" 
G.L3 = ggplot(df, aes(x=expression.level, 
                      y=levels.combined, 
                      fill=multCompLetters)) + 
  geom_density_ridges2(scale=2.5, quantile_lines=TRUE,
                       quantile_fun=function(x,...) mean(x))  + 
  ggtitle("ANOVA followed by Tukey Honest Significant Differences (multicomparison correction)") +
  ylab("Expression/ELT-2 binding") + 
  xlab("L3 intestine expression level") 


df.L3 = df
tbl = table(df.L3$levels.combined) 
tbl = tbl[tbl!=0]
Ns = data.frame(group=names(tbl), val=c(tbl), x=18.5, y=1:length(tbl) + .25)

G.L3 = G.L3 +
  geom_text(inherit.aes = F, data=Ns, aes(y=y, x=x, label=paste("n",Ns$val,sep="=")))

G.L3
ggsave("L3.rlogc.ridgeplot.pdf")
```
```{r paste-prev-plots, eval=FALSE, include=FALSE}
library(grid)
grid.newpage()
grid.draw(cbind(
  ggplotGrob(G.LE 
               ),
  ggplotGrob(G.L1 + 
               ylab(element_blank()) + 
               theme(axis.text.y = element_blank())
             ),
  ggplotGrob(G.L3 + 
               ylab(element_blank()) + 
               theme(axis.text.y = element_blank()))))
ggsave("ridgeplot.pdf")
```

```{r L1-adding-dineenstatus}
data = alldata
data %<>% filter(!is.na(din.status.description))
logFC = data$log2FoldChange.L1
L1_bound = ifelse(data$L1_bound, "ELT2 BOUND", "ELT2 UNbound")
rlogc.L1 = data$rlogc.L1
L1_int_exp = data$L1_int_exp
L1_int_exp[ L1_int_exp == "ftr" ] = "not passing"
din.status.description = data$din.status.description
L1.aov = aov(logFC ~ L1_bound *  din.status.description)
L1.hsd = TukeyHSD(L1.aov, ordered=T, conf.level = .99)
opar = par(mar=c(3,15,2,1)); plot(L1.hsd, las=2,cex.axis=.5); par(opar)
df = data.frame(expression.FoldChange = logFC, 
                L1_bound = L1_bound, 
                intestine_enriched=L1_int_exp,
                levels.combined = paste(L1_bound, din.status.description, sep=", "),
                din.status.description = din.status.description
                )
table(df$levels.combined)

# get the letters

comp = do.call(compare, list(L1.hsd$`L1_bound:din.status.description`[,4],.05))
comp[is.na(comp)] <- FALSE
letters = multcompLetters(comp)$Letters

names(letters) <- sub(":", ", ", names(letters))
df$multCompLetters = letters[as.character(df$levels.combined)]

# order by mean
lvls = df %>% 
  group_by(levels.combined) %>% 
  summarize(mean=mean(expression.FoldChange,na.rm=T)) %>% 
  dplyr::arrange(mean) %>% pull(levels.combined)

df$levels.combined = factor(df$levels.combined, levels=lvls)

G.L1.dineen= ggplot(df %>% filter(!is.na(levels.combined)), aes(y=expression.FoldChange, 
                      x=levels.combined,fill=multCompLetters)) + 
  geom_violin() +
  stat_summary( fun="mean",
               geom = "crossbar", 
               width = 0.5) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 
  
G.L1.dineen
```

```{r norm}


```

```{r ggqq}

ggqqplot(df, x="expression.FoldChange") + facet_wrap(~L1_bound+din.status.description) + ggtitle("Normality (qqplots) of different subsets")

```


```{r subgroups-normality-ks}
dfl = split(df, df$levels.combined)
lapply(dfl, function(x){
  kst = ks.test(x$expression.FoldChange,"pnorm")
  kst
})
```


```{r L3-adding-dineenstatus}
din.status.description = alldata$din.status.description
din.status.description[is.na(din.status.description)] = 'unchanged_ELT2_minus'

L3.aov = aov(rlogc.L3 ~ L3_bound * L3_int_exp * din.status.description)
L3.hsd = TukeyHSD(L3.aov, conf.level = .99)
opar = par(mar=c(3,15,2,1)); plot(L3.hsd, las=2,cex.axis=.5); par(opar)
df = data.frame(expression.level = rlogc.L3, 
                L3_bound = L3_bound, 
                intestine_enriched=L3_int_exp,
                levels.combined = paste(L3_bound, L3_int_exp, din.status.description, sep=", "))
table(df$levels.combined)

# get the letters
comp = do.call(compare, list(L3.hsd$`L3_bound:L3_int_exp:din.status.description`[,4],.01))
comp[is.na(comp)] <- FALSE

letters = multcompLetters(comp)$Letters

names(letters) <- gsub(":", ", ", names(letters))

df$levels.combined = factor(df$levels.combined,
levels = rev(c(
"ELT2 BOUND, equal, down_ELT2_minus"  ,          
"ELT2 BOUND, equal, unchanged_ELT2_minus"     ,   
"ELT2 BOUND, equal, up_ELT2_minus"     ,    

"ELT2 UNbound, equal, unchanged_ELT2_minus"   ,   
"ELT2 UNbound, equal, up_ELT2_minus"     ,  

"ELT2 BOUND, enriched, down_ELT2_minus",          
"ELT2 BOUND, enriched, unchanged_ELT2_minus"    , 
"ELT2 BOUND, enriched, up_ELT2_minus"    , 

"ELT2 UNbound, enriched, down_ELT2_minus",        
"ELT2 UNbound, enriched, unchanged_ELT2_minus" ,  
"ELT2 UNbound, enriched, up_ELT2_minus"  ,  

        
"ELT2 BOUND, not passing, down_ELT2_minus"  ,     
"ELT2 BOUND, not passing, unchanged_ELT2_minus"  ,
"ELT2 BOUND, not passing, up_ELT2_minus"   , 

"ELT2 UNbound, not passing, down_ELT2_minus" ,  
"ELT2 UNbound, not passing, unchanged_ELT2_minus",
"ELT2 UNbound, not passing, up_ELT2_minus"  ,     

"ELT2 BOUND, depleted, down_ELT2_minus",
"ELT2 BOUND, depleted, unchanged_ELT2_minus"  ,  
"ELT2 BOUND, depleted, up_ELT2_minus"      ,     


"ELT2 UNbound, depleted, down_ELT2_minus", 
"ELT2 UNbound, depleted, unchanged_ELT2_minus",
"ELT2 UNbound, depleted, up_ELT2_minus"       
   )))


df$multCompLetters = letters[as.character(df$levels.combined)]

table(df$multCompLetters)

G.L3.dineen= ggplot(df, aes(x=expression.level, 
                      y=levels.combined, fill=multCompLetters)) + 
  geom_density_ridges2(scale=2, quantile_lines=TRUE,
                       quantile_fun=function(x,...) mean(x)) + facet_wrap(~intestine_enriched)
G.L3.dineen
```
